packages/docusaurus-utils/src/__tests__/markdownUtils.test.ts

```diff
@@ -130,6 +130,26 @@ describe('createExcerpt', () => {
     );
   });
 
+  it('creates excerpt for content with imports/exports declarations, with CRLF line endings', () => {
```

Comment:
isn't this test supposed to fail if I remove your change in createExcerpt ? Because it seems to pass before/after in all cases so not sure the test is testing anything 😅

Reply:
Ah ok, ou said one of the fix was not required for the fix 😅 but if I don't revert both of them, the test will pass in both cases. So the 2nd change is required in the end? not sure to understand why

Reply:
Sorry I wasn’t clear 😅 , let me try to rephrase:

The problem is that on Windows, line endings are `\r\n` by default (CRLF). In `createExcerpt`, this caused an issue if the file contained `import` lines, as imports are supposed to end with an empty line. But the condition `fileLine === ''` couldn’t be `true` here, as _empty_ lines actually contained `\r`.

1. First way to fix this problem: [changing the initial _line split_, by splitting on `\r?\n` instead of `\n`](https://github.com/facebook/docusaurus/pull/8606/files#diff-5df6dbf86e88f60f50c8864687bdb2cd37aec3d6a5d05afa6c8fd8dc6327c04dR59).

2. Another way to fix the issue is doing it later, when handling the `import` lines, by [checking for `line.trim()` instead of just `line`](https://github.com/facebook/docusaurus/pull/8606/files#diff-5df6dbf86e88f60f50c8864687bdb2cd37aec3d6a5d05afa6c8fd8dc6327c04dR66), which would remove the extra `\r`. This fix is not needed if we apply the fix 1. already. But as we already check `line.trim()` later in the function for other purpose, I think it makes the check more consistent.

Only one of these fixes is necessary to handle CRLF properly in `createExcerpt`. To me the fix 1 is the cleanest, but it doesn’t hurt to keep the fix 2. too.

If you revert one of these fixes, the test should still pass. If you revert both, the test should fail.

Reply:
I see thanks

Didn't know trim actually removed new lines too 😄 so we have a double solution here, that's why I was a bit confused